Store creation: Remove outdated feature flags and enable store creation by default

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

⚠️ Please make sure that #10360 is reviewed and merged first.
Closes: #10327 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR concludes the code cleanup for store creation. Since we decided to remove the web view solution for store creation, the feature flag `storeCreationM2WithInAppPurchasesEnabled` should be removed along with `storeCreationM2`. Store creation is enabled by default with a single code path for free trial.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Build, run, and log out of the app if needed.
- On the prologue screen, notice that both CTA is available as before: Log In and Get Started.
- Select Get Started and tap Log In.
- Enter an email address that is not associated to any WPCom account.
- See the error screen about incorrect email address > the CTA to Create an account should be available.

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/720b900a-b7b2-49e4-8172-d117011de776" width=320 />

- Navigate back and enter the correct email address to finish login.
- After successfully logging in, switch to the Menu tab and open the site picker.
- Select Add a store > Create a new store.
- The store creation flow should work as before.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
https://github.com/woocommerce/woocommerce-ios/assets/5533851/f4d950d3-b61c-4c16-abb1-7396c891680a

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Remove feature flags storeCreationM2 and storeCreationM2WithInAppPurchasesEnabled

itsmeichigo committed:
Fix test build failure

itsmeichigo committed:
Remove outdated unit test

itsmeichigo committed:
Update release notes

selanthiraiyan reviewed (approved):
> Enter an email address that is not associated to any WP site.

- I think you meant entering an email address that doesn't have a WPCOM account. In that case everything works as expected. 🚀 
- Because, if user enters a WPCOM email address which doesn't have stores we will present the store creation flow.

itsmeichigo committed:
Remove WebPurchasesForWPComPlans and tests

itsmeichigo committed:
Merge branch 'trunk' into feat/10327-remove-stale-feature-flags

itsmeichigo commented:
> I think you meant entering an email address that doesn't have a WPCOM account. In that case everything works as expected

Oh that's correct - I updated the testing steps - thanks for pointing that out!

I also removed the unused `WebPurchasesForWPComPlans` and its tests in ab86a8d18eeab7252a83288cecd9e3d5221b971b as suggested in #10360. Will merge this when CI passes.
